attributes/defaults.rb

```diff
@@ -54,6 +66,26 @@
   default['mysqld']['my.cnf']['mysqld_safe']['skip_log_error'] = true
 end
 
+# Password columns for user passwords
+case node['mysqld']['db_install']
+when 'mariadb'
+  default['mysqld']['pwd_col'] = 'Password'
+when 'percona'
+  if node['mysqld']['repository']['percona']['version'] == '5.6'
+    default['mysqld']['pwd_col'] = 'Password'
+  else
+    default['mysqld']['pwd_col'] = 'authentication_string'
+  end
+when 'mysql'
+  if node['platform'] == 'debian' && node['platform_version'].to_f < 9.0
+    default['mysqld']['pwd_col'] = 'Password'
+  else
+    default['mysqld']['pwd_col'] = 'authentication_string'
+  end
+else
+  default['mysqld']['pwd_col'] = 'authentication_string'
```

Comment:
Is the `else` really required? Unless `mariadb`, `percona` or `mysql` was set, we're going to have some trouble later on anyway, won't we?